src/interfaces.ts

```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
+   */
+  subVerifier?: string;
+
+  /**
+   * Extra verifier params in case of a WebAuthn verifier type.
+   */
+  extraVerifierParams?: WebAuthnExtraParams;
+
+  /**
+   * Any additional parameter (key value pair) you'd like to pass to the login function.
+   */
+  additionalParams?: ExtraParams;
+}
+
 // TODO extend interface and type documentation wherever necessary. @Yash
 export interface ICoreKit {
-  /** The tKey instance, if initialized. */
+  /** The tKey SDK instance
+   *  TKey is the core module on which this wrapper SDK sits for easy integration.
+   **/
   tKey: ThresholdKey | undefined;
 
   // TODO document errors across all interface methods! maybe even define error
   // codes and document which are thrown. in particular here, error is thrown if
   // not factor key is given (either as function parameter or through local
   // storage)
-  // TODO login with id token @Yash
   /**
-   * Login to tKey and initialize all relevant components.
-   * @param loginParams - TKey login parameters.
+   * Login into the SDK in an implicit flow and initialize all relevant components.
+   * @param loginParams - Parameters for Implicit Login.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
    * @returns A Web3 provider if we are not in redirect mode.
    */
   login(loginParams: LoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
 
+  /**
+   * Login into the SDK using ID Token based login and initialize all relevant components.
+   * @param idTokenLoginParams - Parameters with ID Token based Login.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
+   * @returns A Web3 provider
+   */
+  loginWithIdToken(idTokenLoginParams: IdTokenLoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
+
   /**
    * Handle redirect result after login.
-   * @param factorKey - An optional factor key to use at initialization.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
    * @returns A Web3 provider.
    */
   handleRedirectResult(factorKey?: BN): Promise<SafeEventEmitterProvider>;
 
   /**
    * Indicates whether there is an existing session that can be resumed.
+   * @returns A boolean `true` indicating that there is an existing session that can be resumed.
```

Comment:
Ideally comments should wrap at 80 chars.
You can use VSCode extension `rewrap` for this.